components/layout_thread/lib.rs

```diff
@@ -1200,8 +1200,9 @@ impl LayoutThread {
         let ua_stylesheets = &*UA_STYLESHEETS;
         let ua_or_user_guard = ua_stylesheets.shared_lock.read();
         let guards = StylesheetGuards {
-            author: &author_guard,
+            author: &document_guard,
```

Comment:
I think this doesn't need another guard. `author` is the origin of the stylesheets added by the author, so we can just use it instead of renaming it, can't we?

Reply:
The `document_guard` comes from the `style_shared_lock` on the document, and is used for both stylesheets added by the author as well as (now) the registered property set. If you think renaming the field on `StylesheetGuards` from `author` to `document` would make more sense (but that seems to be crossing the abstraction barrier) then I can do that.

Reply:
I think I'd prefer it, yeah. Author sheets are also document sheets, so it doesn't sound really bad.